src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -40,18 +40,29 @@ class DoctrineChoiceLoader implements ChoiceLoaderInterface
      * passed which optimizes the object loading for one of the Doctrine
      * mapper implementations.
      *
-     * @param ObjectManager              $manager      The object manager
-     * @param string                     $class        The class name of the loaded objects
-     * @param IdReader                   $idReader     The reader for the object IDs
-     * @param EntityLoaderInterface|null $objectLoader The objects loader
+     * @param ObjectManager              $manager           The object manager
+     * @param string                     $class             The class name of the loaded objects
+     * @param IdReader                   $idReader          The reader for the object IDs
+     * @param EntityLoaderInterface|null $objectLoader      The objects loader
+     * @param bool                       $checkOptimization For BC, to be removed in 5.0
      */
-    public function __construct(ObjectManager $manager, string $class, IdReader $idReader = null, EntityLoaderInterface $objectLoader = null)
+    public function __construct(ObjectManager $manager, string $class, IdReader $idReader = null, EntityLoaderInterface $objectLoader = null, bool $checkOptimization = true)
```

Comment:
I think we should not expose the `$checkOptimization` argument here (it's only for internal use) but use `func_num_args()`/`func_get_arg()` instead.

Reply:
Done, thanks!